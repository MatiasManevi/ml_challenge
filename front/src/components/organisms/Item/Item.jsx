import React from 'react';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';

import './Item.scss';
import { Price } from 'components/atoms';

const Item = ({ item }) => {
	return (
		<Paper className="item">
			<div className="item__thumbnail">
				<img src={item.picture} alt={`imagen de ${item.title}`} />
			</div>
			<div className="item__content">
				<div className="item__description">
					<Grid container alignItems="baseline">
						<Grid xs={10} item>
							<Price price={item.price} freeShipping={item.free_shipping} />
						</Grid>
						<Grid xs={2} item>
							<p className="item__place">{item.condition}</p>
						</Grid>
					</Grid>
				</div>
				<p className="item__title">{item.title}</p>
			</div>
		</Paper>
	);
};

export default Item;
