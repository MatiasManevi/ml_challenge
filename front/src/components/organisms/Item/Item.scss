@import 'assets/scss/variables';

.item {
	display: flex;
	padding: $space-16;
	justify-content: space-between;
}

.item__thumbnail {
	width: $thumbnail-width;
	height: $thumbnail-height;
	border-radius: $thumbnail-radius;
}

.item__content {
	padding: $space-16;
	width: 100%;
	display: flex;
	flex-direction: column;
}

.item__description {
	display: flex;
	justify-content: space-between;
	align-items: baseline;
	margin-bottom: $space-32;
}

.item__place {
	color: $color-neutral;
	font-size: $font-size-12;
}

.item__title {
	font-size: $font-size-18;
}
