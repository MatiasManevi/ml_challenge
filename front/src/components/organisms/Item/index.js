export { default } from 'components/organisms/Item/Item';
