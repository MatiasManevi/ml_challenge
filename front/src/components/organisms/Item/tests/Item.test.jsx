import React from 'react';
import { shallow } from 'enzyme';

import Item from '../Item';
import item from './item.fixtures';

describe('Item', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<Item item={item} />);
		expect(wrapper).toMatchSnapshot();
	});
});
