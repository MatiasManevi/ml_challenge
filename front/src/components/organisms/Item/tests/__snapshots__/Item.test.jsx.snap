// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="item"
>
  <div
    className="item__thumbnail"
  >
    <img
      alt="imagen de Paleta Padel Wilson Ultimo Modelo Oferta En Toda La Linea Local Baires Deportes Oeste G B A Distribuidor Oficial"
      src="http://http2.mlstatic.com/D_795607-MLA45583919609_042021-O.jpg"
    />
  </div>
  <div
    className="item__content"
  >
    <div
      className="item__description"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="baseline"
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={10}
        >
          <Price
            freeShipping={true}
            price={
              Object {
                "amount": 9950,
                "currency": "$",
                "decimals": "00",
              }
            }
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={2}
        >
          <p
            className="item__place"
          >
            Nuevo
          </p>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </div>
    <p
      className="item__title"
    >
      Paleta Padel Wilson Ultimo Modelo Oferta En Toda La Linea Local Baires Deportes Oeste G B A Distribuidor Oficial
    </p>
  </div>
</WithStyles(ForwardRef(Paper))>
`;
